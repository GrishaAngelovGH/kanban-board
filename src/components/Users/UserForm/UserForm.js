import { useState, useRef } from "react"

import Button from "react-bootstrap/Button"
import Form from "react-bootstrap/Form"
import FormControl from "react-bootstrap/FormControl"

import userRepository from "persistent/persistentUserRepository"

const fileReader = new FileReader()

const UserForm = ({ onUpdate }) => {
  const [image, setImage] = useState("")
  const [username, setUsername] = useState("")

  const imageInputRef = useRef()

  const handleImageChange = ({ target }) => {
    fileReader.readAsDataURL(target.files[0])
    fileReader.onload = e => {
      setImage(e.target.result)
    }
  }

  const handleUsernameChange = ({ target: { value } }) => {
    setUsername(value)
  }

  const handleSubmit = e => {
    e.preventDefault()
    userRepository.createUser(image, username)

    imageInputRef.current.value = ""
    setImage("")
    setUsername("")

    onUpdate()
  }

  return (
    <div className="mt-3">
      <h3 className="text-center text-secondary bg-secondary-subtle rounded">
        Create New User
      </h3>

      <Form onSubmit={handleSubmit}>
        <Form.Group controlId="avatar" className="mb-2">
          <Form.Label>Upload Avatar (optional)</Form.Label>
          <FormControl size="sm" ref={imageInputRef} type="file" onChange={handleImageChange} accept="image/*" />
        </Form.Group>

        <Form.Group controlId="name" className="mb-2">
          <Form.Label>Enter Name</Form.Label>
          <FormControl size="sm" value={username} onChange={handleUsernameChange} required />
        </Form.Group>

        <Button size="sm" type="submit" variant="secondary" className="w-25">Add</Button>
      </Form>
    </div>
  )
}

export default UserForm