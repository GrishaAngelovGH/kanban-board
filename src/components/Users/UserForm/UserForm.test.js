import { render } from "@testing-library/react"

import UserForm from "./UserForm"

test("should render UserForm component", () => {
  const view = render(<UserForm />)

  expect(view).toMatchSnapshot()
})