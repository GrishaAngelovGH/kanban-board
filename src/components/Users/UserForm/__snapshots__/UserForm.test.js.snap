// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserForm component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-3"
      >
        <h3
          class="text-center text-secondary bg-secondary-subtle rounded"
        >
          Create New User
        </h3>
        <form
          class=""
        >
          <div
            class="mb-2"
          >
            <label
              class="form-label"
              for="avatar"
            >
              Upload Avatar (optional)
            </label>
            <input
              accept="image/*"
              class="form-control form-control-sm"
              id="avatar"
              type="file"
            />
          </div>
          <div
            class="mb-2"
          >
            <label
              class="form-label"
              for="name"
            >
              Enter Name
            </label>
            <input
              class="form-control form-control-sm"
              id="name"
              required=""
              value=""
            />
          </div>
          <button
            class="w-25 btn btn-secondary btn-sm"
            type="submit"
          >
            Add
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-3"
    >
      <h3
        class="text-center text-secondary bg-secondary-subtle rounded"
      >
        Create New User
      </h3>
      <form
        class=""
      >
        <div
          class="mb-2"
        >
          <label
            class="form-label"
            for="avatar"
          >
            Upload Avatar (optional)
          </label>
          <input
            accept="image/*"
            class="form-control form-control-sm"
            id="avatar"
            type="file"
          />
        </div>
        <div
          class="mb-2"
        >
          <label
            class="form-label"
            for="name"
          >
            Enter Name
          </label>
          <input
            class="form-control form-control-sm"
            id="name"
            required=""
            value=""
          />
        </div>
        <button
          class="w-25 btn btn-secondary btn-sm"
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
